ui/src/app/base/components/SegmentToggle/SegmentToggle.tsx

```diff
@@ -0,0 +1,37 @@
+import { Button } from "@canonical/react-components";
+import classNames from "classnames";
+
+type Segment<V> = {
+  title: string;
+  value: V;
+};
+
+type Props<V> = {
+  onSelect: (selected: V) => void;
+  options: Segment<V>[];
+  selected: V;
+};
+
+const SegmentToggle = <V,>({
+  onSelect,
+  options,
+  selected,
+}: Props<V>): JSX.Element => {
+  return (
+    <div className="p-segment-toggle">
+      {options.map(({ title, value }) => (
+        <Button
+          className={classNames("p-segment-toggle__button", {
+            "is-active": value === selected,
+          })}
+          key={title}
+          onClick={() => onSelect(value)}
+        >
+          {title}
+        </Button>
+      ))}
+    </div>
```

Comment:
We already have a Tabs component in Vanilla that seems to do the same thing https://vanillaframework.io/docs/patterns/tabs#default

Could you use the vanilla copmonent instead and follow accessibility recommendations (add roles `tablist`, `tab`, `aria-selected` to the currently active tab).

More on this here:
 https://vanillaframework.io/docs/patterns/tabs#accessibility

Reply:
Sure, that's even better. Thanks!